### PR TITLE
SQL: entity in group clause

### PR DESCRIPTION
**Pull request by unrealwork**
## SQL: entity in group clause
- [x] Added tests for case: entity in group clause

### CODE REVIEW

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+        public <T> Response createOrReplace(T query) {
+            Response response = httpEntityGroupApiResource
+                    .request()
+                    .put(Entity.json(query));
+            response.bufferEntity();
+            return response;
+        }
+
+        public <T> Response createOrReplace() {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: I think it's redundant, let user to create default entity by `createOrReplace(new Entity())` if required

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    public static class EntityGroupMethod {
+        private WebTarget httpEntityGroupApiResource;
+        private WebTarget httpEntityGroupEntitiesApiResource;
+
+        EntityGroupMethod(String entityGroupName) {
+            httpEntityGroupApiResource = httpEntitiesApiResource.path(entityGroupName);
+            httpEntityGroupEntitiesApiResource = httpEntityGroupApiResource.path("entities");
+        }
+
+        public <T> Response update(T query) {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: Are you able to call `update(query, "*", new HashMap())` or something like this here?

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+            Response response = httpEntityGroupApiResource
+                    .queryParam("expression", expression)
+                    .queryParam("tags", tags)
+                    .request()
+                    .method("PATCH", Entity.json(query));
+            response.bufferEntity();
+            return response;
+        }
+
+        public <T> Response update(T query, String expression) {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: Are you able to call update(query, "*", new HashMap()) or something like this here?

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+            Response response = httpEntityGroupApiResource
+                    .queryParam("expression", expression)
+                    .request()
+                    .method("PATCH", Entity.json(query));
+            response.bufferEntity();
+            return response;
+        }
+
+        public <T> Response update(T query, Map<String, String> tags) {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: Are you able to call update(query, "*", new HashMap()) or something like this here?

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+            Response response = httpEntityGroupApiResource
+                    .queryParam("expression", expression)
+                    .queryParam("tags", tags)
+                    .request()
+                    .put(Entity.json(query));
+            response.bufferEntity();
+            return response;
+        }
+
+        public <T> Response createOrReplace(T query, String expression) {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: use existing method `createOrReplace(T query, String expression, Map<String, String> tags)`

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+            Response response = httpEntityGroupApiResource
+                    .queryParam("expression", expression)
+                    .request()
+                    .put(Entity.json(query));
+            response.bufferEntity();
+            return response;
+        }
+
+        public <T> Response createOrReplace(T query, Map<String, String> tags) {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: use existing method `createOrReplace(T query, String expression, Map<String, String> tags)`

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+            Response response = httpEntityGroupApiResource
+                    .queryParam("tags", tags)
+                    .request()
+                    .put(Entity.json(query));
+            response.bufferEntity();
+            return response;
+        }
+
+        public <T> Response createOrReplace(T query) {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: use existing method `createOrReplace(T query, String expression, Map<String, String> tags)`

Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: It'is not the same

File: src/test/java/com/axibase/tsd/api/model/entitygroup/EntityGroup.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        return this;
+    }
+
+    public Map<String, String> getTags() {
+        return tags;
+    }
+
+    public EntityGroup setTags(Map<String, String> tags) {
+        this.tags = tags;
+        return this;
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/model/entitygroup/EntityGroup.java]: please don't use chain-notation for consistency

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+                .queryParam("tags", tags)
+                .queryParam("expression", expression)
+                .request()
+                .get();
+        response.bufferEntity();
+        return response;
+    }
+
+    public static Response list(String expression) {
+        Response response = httpEntitiesApiResource
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: use `list(String expression, Map<String, String> tags)` instead

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+                .queryParam("expression", expression)
+                .request()
+                .get();
+        response.bufferEntity();
+        return response;
+    }
+
+    public static Response list(Map<String, String> tags) {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: use `list(String expression, Map<String, String> tags)` instead

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+        Response response = httpEntitiesApiResource
+                .queryParam("tags", tags)
+                .request()
+                .get();
+        response.bufferEntity();
+        return response;
+    }
+
+    public static Response list() {
+        Response response = httpEntitiesApiResource
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: use `list(String expression, Map<String, String> tags)` instead

**Comment by unrealwork** [comment]
fix after code review

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+        Response response = httpEntitiesApiResource.path(entityGroup)
+                .queryParam("expression", expression)
+                .queryParam("tags", tags)
+                .request()
+                .method("PATCH", Entity.json(query));
+        response.bufferEntity();
+        return response;
+    }
+
+    public static <T> Response updateEntityGroup(String entityGroup, T query, String expression) {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: reuse existing functions

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+        Response response = httpEntitiesApiResource.path(entityGroup)
+                .queryParam("expression", expression)
+                .request()
+                .method("PATCH", Entity.json(query));
+        response.bufferEntity();
+        return response;
+    }
+
+    public static <T> Response updateEntityGroup(String entityGroup, T query, Map<String, String> tags) {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: reuse existing functions

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+        Response response = httpEntitiesApiResource.path(entityGroup)
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: reuse existing functions

File: src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+                .queryParam("expression", expression)
+                .queryParam("tags", tags)
+                .request()
+                .method("PATCH", Entity.json(query));
+        response.bufferEntity();
+        return response;
+    }
+
+    public static <T> Response updateEntityGroup(String entityGroup, T query, String expression) {
+        Response response = httpEntitiesApiResource.path(entityGroup)
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/entitygroup/EntityGroupsMethod.java]: reuse existing functions

**Comment by Sild** [comment]
lgth

**Comment by rmakulov** [comment]
lgtm

**Pull request merged**